doc/developers/advanced_installation.rst

```diff
@@ -22,127 +33,191 @@ basis to help users test bleeding edge features or bug fixes::
 Building from source
 =====================
 
-In the vast majority of cases, building scikit-learn for development purposes
-can be done with::
+.. _git_repo:
 
-    pip install cython pytest flake8
+Use `Git <https://git-scm.com/>`_ to check out the latest source from the
+`scikit-learn repository <https://github.com/scikit-learn/scikit-learn>`_ on
+`Github <https://github.com>`_ and pip to build the project::
 
-Then, in the main repository::
+    git clone git://github.com/scikit-learn/scikit-learn.git
+    cd scikit-learn
 
+    pip install cython
     pip install --editable .
 
-Please read below for details and more advanced instructions.
+Check that the installed scikit-learn with a version number ending with
+`.dev0`::
+
+    python -c "import sklearn; sklearn.show_versions()"
+
+Run the tests with ``pytest``::
+
+    pip install pytest
+    pytest sklearn
+
+Please read below for more details and :ref:`platform system specific
+instructions <platform_specific_instructions>` (e.g. in case compilation error
+message).
+
+.. note::
+
+    You will have to re-run the ``pip install --editable .`` command every
+    time the source code of a compiled extension is changed (for instance when
+    switching branches or pulling changes from upstream). Compiled extensions
+    are Cython files (ending in `.pyx` or `.pxd`). See :ref:`editable_mode`
+    for more details.
 
 Dependencies
 ------------
 
-Scikit-learn requires:
+Runtime dependencies
+~~~~~~~~~~~~~~~~~~~~
+
+Scikit-learn requires the following dependencies both at build time and at
+runtime:
 
 - Python (>= 3.5),
 - NumPy (>= 1.11),
 - SciPy (>= 0.17),
 - Joblib (>= 0.11).
 
+Those dependencies are **automatically installed by pip and conda** when
+installing scikit-learn or building scikit-learn from source.
```

Comment:
that is only true if you run `pip install -e .`
If you run `python setup.py build_ext -i` or equivalently `make in`, you need to install the dependencies first.

Reply:
The text says `pip` so I guess that's fine

(not sure if your comment came first ^^)

Reply:
Yes we never mention any other way to build scikit-learn in this doc.

Reply:
Maybe
```
Those dependencies are **automatically installed by pip* if they were missing when
building scikit-learn from source.
```
Saying that they will be installed with conda seems incorrect when building from source. They would be when conda installing scikit-learn, but since we are building from source section that part is probably more relevant? Or anyway reformulate this sentence to avoid confusing users on that point.